src/server/mod.rs

```diff
@@ -0,0 +1,74 @@
+use anyhow::{Context, Result};
+use bb8_redis::{
+    bb8::{Pool, PooledConnection},
+    redis::{AsyncCommands, RedisResult},
+    RedisConnectionManager,
+};
+use std::{collections::HashMap, hash::Hash};
+
+use crate::msg::Message;
+
+pub struct Server<C, F>
+where
+    C: ToString,
+    F: Fn(String),
+{
+    pool: Pool<RedisConnectionManager>,
+    channels: HashMap<C, F>,
+}
+
+impl<C, F> Server<C, F>
+where
+    C: ToString + Eq + Hash,
+    F: Fn(String),
```

Comment:
Fn need to accept some sort of request object. the object can then have multiple attributes, like the scheme and the data (other values from the message?)
the data will already be base64 decoded so it's up to the handler to decide what to do with the data based on the schema